apps/extension/src/ui/atoms/onChainIds.ts

```diff
@@ -0,0 +1,127 @@
+import { OnChainIds } from "@talismn/on-chain-id"
+import { api } from "@ui/api"
+import { useEffect } from "react"
+import { AtomEffect, DefaultValue, atom, selectorFamily, useSetRecoilState } from "recoil"
+
+type OnChainIdsStore = {
+  addresses: Map<string, number>
+  onChainIds: OnChainIds
+}
+
+//
+// hooks
+//
+
+export const useFetchOnChainIdsForAddresses = (addresses: string[] | string | undefined) => {
+  const setMounted = useSetRecoilState(fetchAddresses(addresses))
+  useEffect(() => {
+    if (!addresses) return
+    setMounted(true)
+    return () => setMounted(false)
+  }, [addresses, setMounted])
+}
+
+//
+// selectors
+//
+
+export const readOnChainIds = selectorFamily({
```

Comment:
feeling the same with this one, wdyt about replacing it with a useQueryCall ?

Reply:
I like how this one collects all the addresses to be fetched into one place, but I think in the future this should be done inside the `@talismn/on-chain-ids` lib instead of on the frontend